lib/manager/npm/post-update/index.ts

```diff
@@ -336,9 +337,71 @@ async function resetNpmrcContent(
   }
 }
 
+// istanbul ignore next
+async function updateYarnOffline(
+  lockFileDir: string,
+  localDir: string,
+  updatedArtifacts: UpdatedArtifacts[]
+): Promise<void> {
+  try {
+    const resolvedPaths: string[] = [];
+    const yarnrc = await getFile(upath.join(lockFileDir, '.yarnrc'));
+    const yarnrcYml = await getFile(upath.join(lockFileDir, '.yarnrc.yml'));
+
+    if (yarnrc) {
+      // Yarn 1 (offline mirror)
+      const mirrorLine = yarnrc
+        .split('\n')
+        .find((line) => line.startsWith('yarn-offline-mirror '));
+      if (mirrorLine) {
+        const mirrorPath = mirrorLine
+          .split(' ')[1]
+          .replace(/"/g, '')
+          .replace(/\/?$/, '/');
+        resolvedPaths.push(upath.join(lockFileDir, mirrorPath));
+      }
+    } else if (yarnrcYml) {
+      // Yarn 2 (offline cache and zero-installs)
+      const config = parseSyml(yarnrcYml);
```

Comment:
Fwiw, the name is a bit misleading but it's regular YAML, so you can use any parser you want, not necessarily the Yarn one. Or you can use the `Configuration` class from `@yarnpkg/core` (which would take care of the default values, etc), but it's perhaps a bit more complicated than necessary.

Reply:
@arcanis Thank you for your valuable comments!

`parseSyml` is already used to parse the lockfile: https://github.com/renovatebot/renovate/blob/c65d40048936e569c6413092bf22d5e2abe3faf7/lib/manager/npm/extract/yarn.ts#L15 and `@yarnpkg/parsers` seems to use `js-yaml` so I think it's OK and would give more consistent behavior with Yarn, e.g., empty file handling.